test/websocket.misc.js

```diff
@@ -0,0 +1,37 @@
+var assert = require('assert');
+var utils = require('./helpers/test.utils');
+var Web3 = utils.getWeb3();
+
+describe('Websocket etc & errors', function () {
+    let web3;
+
+    beforeEach(() => web3 = new Web3('ws://localhost:' + 8777));
+
+    // This test's error is fired by the request queue checker in the onClose handler
+    it('errors when there is no connection', async function(){
+        try {
+            await web3.eth.getBlockNumber();
+            assert.fail();
+        } catch (err) {
+            assert(err.message.includes('connection not open'));
+        }
+    });
+
+    // This test's error is fired by the readyState check in .send
+    it('errors when requests continue after socket closed', async function(){
+        try { await web3.eth.getBlockNumber() } catch (err) {
```

Comment:
Is this try/catch required?

Reply:
Yes, see above. There are two different locations the error is thrown from. The first failure is thrown from `_onClose`, the second from `send.`

Reply:
Thanks for your explanation! Could you add such a note as an in-line comment to the related test case? (A future developer will definitely like such a hint) :)

Reply:
Add more comments in 4044687b2992782eb2dffea26014a7f9ca56ef75